core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -149,15 +149,13 @@ func (s *onchainSubscriptions) queryLoop() {
 		}
 
 		if start > lastKnownCount {
+			updateLastKnownCount()
 			start = 1
 		}
 
-		end := start + uint64(s.config.UpdateRangeSize)
+		end := start + uint64(s.config.UpdateRangeSize) - 1
 		if end > lastKnownCount {
-			updateLastKnownCount()
-			if end > lastKnownCount {
-				end = lastKnownCount
-			}
+			end = lastKnownCount
 		}
 		if err := s.querySubscriptionsRange(ctx, blockNumber, start, end); err != nil {
```

Comment:
Can we add a new debug-level log inside querySubscriptionsRange() to always log the range that actually got updated?